### PR TITLE
Feature/ui wizard create stream

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds following components,
  - `Wizard` - A wrapper that connects everything that we call as wizard
  - `WizardModal` - A wrapper for dropping in the modal anywhere we want. A generic re-usualble way to show modal(wizards) anywhere in a more consistent way.
  - `ResourceCenter` - Nothing fancy as of now. Just shows a dummy list of entities that we can create in CDAP
  - `ResourceCenter Entity` - A short info on the entity that is being created + a create link (to be used in the resource center)
  - `TimeToLive` - A naive implentation of setting ttl for streams in UI (Need some inputs)
  - `SelectWithOptions` - One element to render a dropdown instead of nesting `<Input>` with `<options>` like the one from reactstrap. This is specifically useful when we use `connect` from `react-redux`
  - `SimpleSchema` - A naive implementation of a simple schema editor. Nothing fancy. A table of textbox, dropdown, checkbox & a delete button (just like the way we have in hydrator).
- Adds following services,
  - `WizardConfigs` - One place to find all WizardConfigurations that are being used in CDAP
  - `WizardStores` - Once place to find all stores used by each Wizard instances. 
- Have added aliases to webpack to refer to any component/service from anywhere in the application (could avoid using relative paths with a lot of `../.`)
- Have added an optional field in `ui-settings.json` to enable debugging in UI (console log in the browser).
#### Note:
- TTL is not yet integrated with backend.
- Haven't added the functionality of disabling steps after finishing creating a stream
- Working on i18n and will add as separate PR as this has already exceeded the limit we can review.
- Upload data coming soon.

### CODE REVIEW

File: cdap-ui/app/cdap/components/CaskWizards/StreamCreate/SchemaStep.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          type: CreateStreamActions.setSchema,
+          payload: { schema }
+        });
+      }
+    };
+  }
+)(SimpleSchema);
+
+export default function SchemaStep() {
+  console.log('asdasd', CreateStreamStore);
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/CaskWizards/StreamCreate/SchemaStep.js:63]: remove

**Review by elfenheart — CHANGES_REQUESTED** [review]

**Comment by ajainarayanan** [comment]
Closing the PR and opening the same against feature/cdap-4-sneak-peek

File: cdap-ui/app/cdap/api/stream/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Datasource from '../../services/datasource';
+import {apiCreator} from '../../services/resource-helper';
+
+let dataSrc = new Datasource();
+let basepath = '/namespaces';
```
Comment (elfenheart) [inline cdap-ui/app/cdap/api/stream/index.js]: we can just change this basepath to `/namespaces/:namespace/streams`

**Review by elfenheart — APPROVED** [review]

**Comment by elfenheart** [comment]
👍  LGTM

Don't forget to squash

**Comment by ajainarayanan** [comment]
Have squashed the commits and hence merging.

**Pull request merged**